runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PTransformTranslation.java

```diff
@@ -461,6 +475,38 @@ public RunnerApi.PTransform translate(
           }
         }
       }
+
+      Row configRow = null;
+      try {
+        configRow = payloadTranslator.toConfigRow(appliedPTransform.getTransform());
+      } catch (UnsupportedOperationException e) {
+        // Optional toConfigRow() has not been implemented. We can just ignore.
+      } catch (Exception e) {
+        LOG.warn(
+            "Could not attach the config row for transform "
+                + appliedPTransform.getTransform().getName()
+                + ": "
+                + e);
+        // Ignoring the error and continuing with the translation since attaching config rows is
+        // optional.
+      }
+      if (configRow != null) {
+        ByteStringOutputStream rowOutputStream = new ByteStringOutputStream();
+        try {
+          RowCoder.of(configRow.getSchema()).encode(configRow, rowOutputStream);
+        } catch (IOException e) {
+          throw new RuntimeException(e);
+        }
+        transformBuilder.putAnnotations(CONFIG_ROW_KEY, rowOutputStream.toByteString());
+
+        ByteStringOutputStream schemaOutputStream = new ByteStringOutputStream();
+        try (ObjectOutputStream schemaObjOut = new ObjectOutputStream(schemaOutputStream)) {
+          schemaObjOut.writeObject(SchemaTranslation.schemaToProto(configRow.getSchema(), true));
```

Comment:
SchemaTranslation.schemaToProto is a Proto object. Just use `toByteArray` to serialize this proto. You can use ByteString.copyFrom to go from byte[] to ByteString and not have to deal with streams at all.